api/src/main/java/uk/gov/dvsa/mot/app/TradeServiceRequestHandler.java

```diff
@@ -268,4 +269,55 @@ public Vehicle getLatestMotTest(TradeServiceRequest request, Context context) th
             throw new InternalServerErrorException(e, context.getAwsRequestId());
         }
     }
+
+    /**
+     * Get the latest vehicle information and MOT test recorded for the vehicle specified by MOT Test Number.
+     *
+     * @param request Describes the request. For this method, we require only the registration as a path parameter.
+     * @param context AWS Lambda request context
+     * @return The vehicle and its latest MOT, if found.
+     * @throws TradeException If there is a retrieval error or the vehicle and test are not found.
+     */
+    public Vehicle getLatestMotTestByMotTestNumber(TradeServiceRequest request, Context context) throws TradeException {
+
+        try {
+            logger.trace("Entering getLatestMotTestByMotTestNumber");
+            if (request.getPathParams().getNumber() != null) {
```

Comment:
Instead of calling request.getPathParams().getNumber() multiple times should we call it once and save to a variable